spec/api-system-preferences-spec.js

```diff
@@ -35,6 +35,54 @@ describe('systemPreferences module', () => {
     })
   })
 
+  describe('systemPreferences.registerDefaults(defaults)', () => {
+    before(function () {
+      if (process.platform !== 'darwin') this.skip()
+    })
+
+    it('registers defaults', () => {
+      // to pass into registerDefaults
+      const defaultsDict = {
+        'one': 'ONE',
+        'two': 2,
+        'three': [1, 2, 3]
+      }
+
+      // to test type
+      const defaultsMap = [
+        { key: 'one', type: 'string', value: 'ONE' },
+        { key: 'two', value: 2, type: 'integer' },
+        { key: 'three', value: [1, 2, 3], type: 'array' }
+      ]
+
+      systemPreferences.registerDefaults(defaultsDict)
+
+      for (const def of defaultsMap) {
```

Comment:
sorry for nitpicking, but `def` isn't a word )

actually you can make the destructuring right here to avoid naming a variable at all (isn't it cool?!)

```js
for (const {key, value: expectedValue, type} of defaultsMap) {
  const actualValue = systemPreferences.getUserDefault(key, type)
  assert.deepEqual(actualValue, expectedValue)
}
```

Reply:
ooh wait that is cool...totally didn't know that

Reply:
I didn't know that either, that is cool!

Although, I'm not sure I see the value of these temporaries over something like

```js
for (const test of tests) {
  const actual = systemPreferences.getUserDefault(test.key, test.type)
  assert.deepEqual(actual, test.expected)
}
'''

Reply:
If there is no test data and test logic duplication, choosing one option over the other is just a matter of a personal preference.

Reply:
Agreed that use of `test.key` or `key` is a preference rather than a functional difference